### PR TITLE
[AMORO-3104] Added table summary metrics

### LINKED ISSUE #3104: [Feature]: Add table summary metrics
### Description

Add table summary metrics for each table to help users better understand the detail of tables.

### Use case/motivation

Amoro has currently implemented the Metric System to provide various metric information externally. 

However, it is also crucial to have metrics for important aspects such as data size, file count, and record numbers related to tables. By adding metrics for these aspects, we aim to assist users in gaining a better understanding of the various situations on the tables.

### Describe the solution

We should define proper table summary metrics first.
Then AMS will refresh the table runtime every 3 minutes(by default), it can update the table summary metrics in this stage.

### Subtasks

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by MarigWeizhi**
## Why are the changes needed?

Close #3104.

## Brief change log

- Added `TableSummaryMetrics` for each table to help users better understand the detail of tables.
- Changed the structure of `OptimizingEvaluator.PendingInput`
- Added a column `table_summary` to table `table_runtime`

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-ams/amoro-ams-server/src/main/resources/postgres/ams-postgres-init.sql [hunk 1]
```diff
@@ -211,6 +212,7 @@ COMMENT ON COLUMN table_runtime.optimizer_group IS 'Optimizer group';
 COMMENT ON COLUMN table_runtime.table_config IS 'Table-specific configuration';
 COMMENT ON COLUMN table_runtime.optimizing_config IS 'Optimizing configuration';
 COMMENT ON COLUMN table_runtime.pending_input IS 'Pending input data';
+COMMENT ON COLUMN table_runtime.table_summary IS 'Table Summary data';
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/resources/postgres/ams-postgres-init.sql:215]:
```suggestion
COMMENT ON COLUMN table_runtime.table_summary IS 'Table summary data';
```

File: amoro-ams/amoro-ams-server/src/main/resources/postgres/upgrade.sql [hunk 2]
```diff
@@ -22,3 +22,7 @@ ALTER TABLE `table_blocker` DROP INDEX `table_index`;
 ALTER TABLE `table_blocker` ADD COLUMN `prev_blocker_id` bigint(20) NOT NULL DEFAULT -1;
 COMMENT ON COLUMN table_blocker.prev_blocker_id IS 'prev blocker id when created';
 ALTER TABLE `table_blocker` ADD UNIQUE KEY `uq_prev` (`catalog_name`,`db_name`,`table_name`, `prev_blocker_id`);
+
+-- ADD COLUMN table_summary FOR TABLE_RUNTIME
+ALTER TABLE table_runtime ADD COLUMN table_summary TEXT;
+COMMENT ON COLUMN table_runtime.table_summary IS 'Table Summary data';
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/resources/postgres/upgrade.sql:28]:
```suggestion
COMMENT ON COLUMN table_runtime.table_summary IS 'Table summary data';
```

**Review by zhoujinsong — COMMENTED** [review]
Thanks for the contribution!

All changes look good to me except a minor issue.

**Comment by MarigWeizhi** [comment]
> Thanks for the contribution!
> 
> All changes look good to me except a minor issue.

fixed

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Comment by zhoujinsong** [comment]
Thanks for the contribution! @MarigWeizhi

**Pull request merged**